clients/interfaces/deviceservicecommand.go

```diff
@@ -17,6 +17,8 @@ import (
 type DeviceServiceCommandClient interface {
 	// GetCommand invokes device service's command API for issuing get(read) command
 	GetCommand(ctx context.Context, baseUrl string, deviceName string, commandName string, queryParams string) (*responses.EventResponse, errors.EdgeX)
+	// GetCommandWithObject invokes device service's get command API and pass the parameters in the payload
+	GetCommandWithObject(ctx context.Context, baseUrl string, deviceName string, commandName string, queryParams string, parameters map[string]interface{}) (*responses.EventResponse, errors.EdgeX)
```

Comment:
Why can't parameters just be `parameters interface{}` and marshal it to JSON for the body

Reply:
Because I leverage the device-skd-go function to prase the request body as `map[string]interface{}`, see the changes:
https://github.com/edgexfoundry/device-sdk-go/pull/1043/files#diff-7143a51f279090b40470ef1214d5ac141c070ff0b0b1275863da6aca8c4da0abL41-R45

And merge the map to request attributes, see the changes:
https://github.com/edgexfoundry/device-sdk-go/pull/1043/files#diff-54ddce28dd2ff1bcd97f51b3bf766407fb0ac86b85861313fbb37a8cfb459b39R140-R206

So the protocol driver can retrieve the parameter from `reqs []sdkModels.CommandRequest)` in `HandleReadCommands` function.
https://github.com/edgexfoundry/device-sdk-go/blob/1be87f15213c05ab1957194b5dea87ff4eff74a4/example/driver/simpledriver.go#L139

For example, the Get Command request contains the payload:
```
curl --location --request GET 'http://0.0.0.0:59882/api/v2/device/name/Simple-Device01/Switch' \
--header 'Content-Type: application/json' \
--data-raw ' {
    "foo":"123",
    "bar":{
        "foobar":"456"
    },
    "anyDataType": ... 
}'
```
The commandRequest should be
```
{
	DeviceResourceName: "..."
	Attributes :  {
                "foo":"123",
                "bar":{
                   "foobar":"456"
                 },
                 "anyDataType": ...
              }
	Type : "..."
}
```